tools/rosbag/include/rosbag/player.h

```diff
@@ -177,17 +178,24 @@ class ROSBAG_DECL Player
 
     void printTime();
 
+    bool pauseCallback(std_srvs::SetBool::Request &req, std_srvs::SetBool::Response &res);
 
 private:
 
     PlayerOptions options_;
 
     ros::NodeHandle node_handle_;
 
+    ros::ServiceServer pause_service_;
```

Comment:
I think ABI would be maintained if all the new fields (and method) were appended to the class rather than inserted.

However, IMO this is not worth it for the loss in source readability— there are unlikely to be external packages linking to this class.